api/v1alpha1/nutanix_node_types.go

```diff
@@ -60,6 +60,25 @@ type NutanixMachineDetails struct {
 	// The project must already be present in the Prism Central.
 	// +kubebuilder:validation:Optional
 	Project *NutanixResourceIdentifier `json:"project,omitempty"`
+
+	// List of GPU devices that need to be added to the machines.
+	// +kubebuilder:validation:Optional
+	GPUs []NutanixGPU `json:"gpus,omitempty"`
+}
+
+type NutanixGPU struct {
+	// Type is the identifier type to use for this resource.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Enum:=deviceID;name
```

Comment:
Thanks for adding the type alias!

I think we might be able to remove the `validation:Enum` tag, because the source type has it.
```suggestion
```